File: codeforces/18.tram/tram.py
# Author: José Rodolfo (jric2002)
n = int(input())
min_capacity = 0
passengers = 0
while(n):
  input_data = input().split(" ")
  a = int(input_data[0])
  b = int(input_data[1])
  passengers = (passengers - a) + b
  if (passengers > min_capacity):
    min_capacity = passengers
  n -= 1
print(min_capacity)

File: codeforces/18.tram/tram.cpp
#include <iostream>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  signed short int n;
  unsigned int min_capacity = 0;
  unsigned int a, b, passengers = 0;
  cin >> n;
  while (n--) {
    cin >> a >> b;
    passengers = (passengers - a) + b;
    if (passengers > min_capacity) {
      min_capacity = passengers;
    }
  }
  cout << min_capacity << endl;
  return 0;
}